pymc/distributions/distribution.py

```diff
@@ -83,6 +85,60 @@
 PLATFORM = sys.platform
 
 
+def filter_shared_RNGs(params):
+    return [p for p in params if not isinstance(p, RandomGeneratorSharedVariable)]
```

Comment:
This should be more general and still apply to your use case.
```suggestion
def filter_RNGs(params):
    return [p for p in params if not isinstance(p.type, RandomType)]
```